### PR TITLE
JobStore as a database entry

### PR DESCRIPTION
**Pull request by cjllanwarne**

### CODE REVIEW

File: engine/src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala [hunk 1]
```diff
@@ -182,7 +182,7 @@ class WorkflowManagerActor(config: Config,
      Watched transitions
      */
     case Event(Transition(workflowActor, _, toState: WorkflowActorTerminalState), data) =>
-      log.info(s"$tag ${workflowActor.path.name} is in a terminal state: $toState")
+      log.debug(s"$tag ${workflowActor.path.name} is in a terminal state: $toState")
```
Comment (mcovarr) [inline engine/src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala]: why isn't this one using the fancy substitution like the other messages?

File: engine/src/main/scala/cromwell/jobstore/SqlJobStore.scala [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        returnCode,
+        Some(jobResult.toJson.toString),
+        None,
+        None)
+      case (key, JobResultFailure(returnCode, throwable, retryable)) => JobStoreEntry(
+        key.workflowId.toString,
+        key.callFqn,
+        key.index.fromIndex,
+        key.attempt,
+        jobSuccessful = true,
```
Comment (mcovarr) [inline engine/src/main/scala/cromwell/jobstore/SqlJobStore.scala]: successful = true?

File: engine/src/main/scala/cromwell/jobstore/SqlJobStore.scala [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        returnCode,
+        None,
+        Some(throwable.getMessage),
+        Some(retryable))
+    }
+  }
+
+  override def readJobResult(jobStoreKey: JobStoreKey)(implicit ec: ExecutionContext): Future[Option[JobResult]] = {
+    sqlDatabase.fetchJobResult(jobStoreKey.workflowId.toString, jobStoreKey.callFqn, jobStoreKey.index.fromIndex, jobStoreKey.attempt) map { _ map { toJobResult } }
```
Comment (mcovarr) [inline engine/src/main/scala/cromwell/jobstore/SqlJobStore.scala]: that innermost pair of curlies looks unnecessary.

**Comment by mcovarr** [comment]
:+1:

File: engine/src/main/scala/cromwell/jobstore/SqlJobStore.scala (hunk 2 above)
Comment (cjllanwarne) [inline engine/src/main/scala/cromwell/jobstore/SqlJobStore.scala]: 😨

Comment (cjllanwarne) [inline engine/src/main/scala/cromwell/jobstore/SqlJobStore.scala]: interesting that you'd point out this one but not comment that the unsuccessful case also still had `successful = true`...

Comment (cjllanwarne) [inline engine/src/main/scala/cromwell/jobstore/SqlJobStore.scala]: wait, what was your comment here?

Comment (cjllanwarne) [inline engine/src/main/scala/cromwell/jobstore/SqlJobStore.scala]: OK I confused myself here. The comment is in a different place in this preview vs the "full code" view. Github bug. My bug is fixed.

File: database/src/main/scala/cromwell/database/slick/JobStoreSlickDatabase.scala [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  override def addJobStoreEntries(entries: Iterable[JobStoreEntry])(implicit ec: ExecutionContext): Future[Unit] = {
+    val action = dataAccess.jobStoreAutoInc ++= entries
+    runTransaction(action) map { _ => () }
+  }
+
+  override def fetchJobResult(workflowUuid: String, callFqn: String, index: Int, attempt: Int)
+                    (implicit ec: ExecutionContext): Future[Option[JobStoreEntry]] = {
+    val action = dataAccess.jobStoreEntriesByJobStoreKey(workflowUuid, callFqn, index, attempt).result
+    runTransaction(action) map {
+      x => if (x.size == 1) Some(x.head)
```
Comment (geoffjentry) [inline database/src/main/scala/cromwell/database/slick/JobStoreSlickDatabase.scala]: Option

**Comment by geoffjentry** [comment]
👍

File: database/src/main/scala/cromwell/database/slick/JobStoreSlickDatabase.scala (hunk 4 above)
Comment (cjllanwarne) [inline database/src/main/scala/cromwell/database/slick/JobStoreSlickDatabase.scala]: I'll change it, but I'll reiterate that I disagree with doing this :)

**Pull request merged**